x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.test.ts

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import {
+  ILegacyClusterClient,
+  KibanaResponseFactory,
+  RequestHandler,
+  RouteConfig,
+} from 'kibana/server';
+import {
+  elasticsearchServiceMock,
+  httpServerMock,
+  httpServiceMock,
+  loggingSystemMock,
+  savedObjectsClientMock,
+} from 'src/core/server/mocks';
+import { parseExperimentalConfigValue } from '../../../../common/experimental_features';
+import { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { EndpointAppContextService } from '../../endpoint_app_context_services';
+import {
+  createMockEndpointAppContextServiceStartContract,
+  createMockPackageService,
+  createRouteHandlerContext,
+} from '../../mocks';
+import { registerHostIsolationRoutes } from './isolation';
+import { createMockConfig } from '../../../lib/detection_engine/routes/__mocks__';
+import { LicenseService } from '../../../../common/license';
+import { Subject } from 'rxjs';
+import { ILicense } from '../../../../../licensing/common/types';
+import { licenseMock } from '../../../../../licensing/common/licensing.mock';
+import { License } from '../../../../../licensing/common/license';
+import {
+  ISOLATE_HOST_ROUTE,
+  UNISOLATE_HOST_ROUTE,
+  metadataTransformPrefix,
+} from '../../../../common/endpoint/constants';
+import {
+  EndpointAction,
+  HostIsolationResponse,
+  HostMetadata,
+} from '../../../../common/endpoint/types';
+import { EndpointDocGenerator } from '../../../../common/endpoint/generate_data';
+import { createV2SearchResponse } from '../metadata/support/test_support';
+import { ElasticsearchAssetType } from '../../../../../fleet/common';
+
+interface CallRouteInterface {
+  body?: any;
+  idxResponse?: any;
+  searchResponse?: HostMetadata;
+  mockUser?: any;
+  license?: License;
+}
+
+const Platinum = licenseMock.createLicense({ license: { type: 'platinum', mode: 'platinum' } });
+const Gold = licenseMock.createLicense({ license: { type: 'gold', mode: 'gold' } });
+
+describe('Host Isolation', () => {
+  let endpointAppContextService: EndpointAppContextService;
+  let mockResponse: jest.Mocked<KibanaResponseFactory>;
+  let licenseService: LicenseService;
+  let licenseEmitter: Subject<ILicense>;
+
+  let callRoute: (
+    routePrefix: string,
+    opts: CallRouteInterface
+  ) => Promise<jest.Mocked<SecuritySolutionRequestHandlerContext>>;
+  const superUser = {
+    username: 'superuser',
+    roles: ['superuser'],
+  };
+
+  const docGen = new EndpointDocGenerator();
+
+  beforeEach(() => {
+    // instantiate... everything
+    const mockScopedClient = elasticsearchServiceMock.createLegacyScopedClusterClient();
```

Comment:
Please considering using elasticsearchServiceMock.createClusterClient(); instead, as we use `asCurrentUser` in the following lines: L132, -L139.